modules/assim.sequential/R/Multi_Site_Constructors.R

```diff
@@ -189,6 +189,8 @@ Construct.H.multisite <- function(site.ids, var.names, obs.t.mean){
     
     if(is.null(obs.names)) next;
     
+    choose <- sapply(var.names, agrep, x = names(obs.t.mean[[site]]),
+                       max = 1, USE.NAMES = F) %>% unlist
```

Comment:
@para2x Is `agrep` instead of `grep` here intentional? `agrep` is "approximate" grep -- i.e. "fuzzy matching", which I am _really_ wary of relying on. At the very least, I would expect a comment here justifying the use of `agrep`.

Also, please don't abbreviate `FALSE` to `F` (@para2x this is primarily on you -- after all, this was modeled after your code). `FALSE` is a reserved keyword, but `F` is not, so it's too easy to accidentally overwrite it in the global environment and wreak havoc on code like this.

Reply:
@ashiklom I'll change the F's to FALSE. That may have just been me. @para2x did the agrep, so I don't know if that was intentional or not. I'm assuming it is just incase there is very slight differences between the state.variables in the SDA xml and the way the variable was named in the R script for obs.mean and obs.cov. (e.g. LAI vs lai)

Reply:
@ashiklom I might have initially written this code somewhere, but this line here was just added by @bailsofhay not me (@bailsofhay's PR).

@bailsofhay, One thing I tried to also bring up in our last meeting was the fact that this way of thinking about the `H`  makes an implicit assumption about the order of variables in cov matrix and secondly I was also wondering if this worked under the years/sites that we have either one not both at the same time ?

Reply:
@para2x Bailey modeled this code after the lines you wrote below (`choose.col`, `choose.row`), which is why I was asking you. I personally think using `agrep` in production code like this is a bad idea -- we should be precise in our definitions and configurations and error quickly and loudly when something isn't right. Trying to predict what users meant to type is very hard and error-prone, and can introduce really insidious bugs (e.g. `WAI` could get corrected to `LAI`, even though they are two very different variables).

Reply:
@ashiklom and @para2x I added that line, but it was stolen from the Construct.R function exactly.... so..... sort of me, but not.

Reply:
@para2x Yea. I get what you're saying. I'm a little confused by what you're saying in the second part. I know when I was testing that function that it appears okay for now because the variables are named for each year at each site in obs.mean, and agrep matches them up to the state.variable names in the XML... so H is correct and in the correct order. i specifically checked for that just in case. H assumes that the way the variables are listed (and their order) is the same as obs.mean and obs.cov.... so as long as obs.cov is built the same way as obs.mean in the script,  H should be okay